modules/ocs2/src/main/scala/gem/ocs2/Parsers.scala

```diff
@@ -539,9 +539,144 @@ object Parsers {
   }
 
   object Gnirs {
-    val wellDepth: PioParse[GnirsWellDepth] = enum(
-      "SHALLOW" -> GnirsWellDepth.Shallow,
-      "DEEP"    -> GnirsWellDepth.Deep
-    )
+
+    val camera: PioParse[GnirsCamera] = {
+
+      import GnirsCamera._
+
+      enum(
+        "short blue" -> ShortBlue,
+        "long blue"  -> LongBlue,
+        "short red"  -> ShortRed,
+        "long red"   -> LongRed
+      )
+    }
+
+    val decker: PioParse[GnirsDecker] = {
+
+      import GnirsDecker._
+
+      enum(
+        "acquisition"            -> Acquisition,
+        "pupil viewer"           -> PupilViewer,
+        "short camera long slit" -> ShortCamLongSlit,
+        "short camera x-disp"    -> ShortCamCrossDispersed,
+        "IFU"                    -> Ifu,
+        "long camera long slit"  -> LongCamLongSlit,
+        "wollaston"              -> Wollaston
+      )
+
+    }
+
+    val disperser: PioParse[GnirsDisperser] = {
+
+      import GnirsDisperser._
+
+      enum(
+        "10 l/mm grating"  -> D10,
+        "32 l/mm grating"  -> D32,
+        "112 l/mm grating" -> D111
+      )
+
+    }
+
+    val disperserOrder: PioParse[GnirsDisperserOrder] = {
+
+      import GnirsDisperserOrder._
+
+      enum(
+        "4.85" -> One,
+        "3.4"  -> Two,
+        "2.2"  -> Three,
+        "1.65" -> FourXD,
+        "1.63" -> Four,
+        "1.25" -> Five,
+        "1.1"  -> Six
```

Comment:
What happened to `Seven` and `Eight`?

Reply:
I forgot them because `Seven` and `Eight` are not selectable through in the OT UI but are present in the `OCS`. I'll check if they are ever used anywhere in the `OCS` codebase. Without the UI label I'm not sure under which exact values they will be exported.